### PR TITLE
Fix incorrect detection of imperative route for async/reactive void methods

### PR DESCRIPTION
**Pull request by dstepanov**

### CODE REVIEW

**Review by timyates — APPROVED** [review]
Using "[The Technique](https://github.com/micronaut-projects/micronaut-core/wiki/Gradle-Build-Tips#building-a-module-against-a-local-version-of-micronaut-core)", I ran this against micronaut-micrometer and it worked https://ge.micronaut.io/s/7hdsrnee7odfy
Also ran it against micronaut-tracing and it fixed that too https://ge.micronaut.io/s/iztalynzws6xa

🦸

File: router/src/main/java/io/micronaut/web/router/DefaultRouteInfo.java [hunk 1]
```diff
@@ -143,7 +143,7 @@ public DefaultRouteInfo(AnnotationMetadata annotationMetadata,
             this.consumesMediaTypes = consumesMediaTypes;
             consumesMediaTypesContainsAll = this.consumesMediaTypes.contains(MediaType.ALL_TYPE);
         }
-        this.imperative = isVoid || !suspended && !reactive && !async && !returnType.getType().equals(Object.class)
+        this.imperative = returnType.getType() == void.class || !suspended && !reactive && !async && !returnType.getType().equals(Object.class)
```
Comment (sdelamo) [inline router/src/main/java/io/micronaut/web/router/DefaultRouteInfo.java:146]: Should we check the `isVoid` as well?

```suggestion
        this.imperative = (isVoid || returnType.getType() == void.class) || !suspended && !reactive && !async && !returnType.getType().equals(Object.class)
```

**Review by graemerocher — APPROVED** [review]

File: router/src/main/java/io/micronaut/web/router/DefaultRouteInfo.java (hunk 1 above)
Comment (dstepanov) [reply router/src/main/java/io/micronaut/web/router/DefaultRouteInfo.java:146]: No, that method includes async reactive methods

Comment (dstepanov) [reply router/src/main/java/io/micronaut/web/router/DefaultRouteInfo.java:146]: *That field

**Comment by timyates** [comment]
@dstepanov There's a couple of test failures https://ge.micronaut.io/s/2tlesgdhatbtm/tests/overview?outcome=FAILED

**Comment by dstepanov** [comment]
Fixed

**Review by graemerocher — APPROVED** [review]

**Pull request merged**